components/eamxx/src/physics/rrtmgp/eamxx_rrtmgp_process_interface.cpp

```diff
@@ -382,6 +402,48 @@ void RRTMGPRadiation::initialize_impl(const RunType /* run_type */) {
   // Determine whether or not we are using a fixed solar zenith angle (positive value)
   m_fixed_solar_zenith_angle = m_params.get<double>("Fixed Solar Zenith Angle", -9999);
 
+  // Not sure what to do about this
+  constexpr int ps = SCREAM_PACK_SIZE;
+
+  // Get time-varying input fields and their associated file(s)
+  // TODO: Harden this impl; get the EKAT msg/warn stuff and use it here to avoid bad logic at the outset
+  // TODO: Maybe also move this to the very top?
+  m_time_varying_active_gases_list =
+      m_params.get<std::vector<std::string>>("time_varying_active_gases_list", {"nolistgiven"});
+  m_time_varying_active_gases_file =
+      m_params.get<std::vector<std::string>>("time_varying_active_gases_file", {"nofilegiven"});
+
+  // Initialize the time interpolator
+  if (m_time_varying_active_gases_file[0] != "nofilegiven") {
+    m_time_interp =
+        util::TimeInterpolation(m_grid, m_time_varying_active_gases_file);
+
+    // To create helper fields for to interpolate their values
+    for (std::string name : m_time_varying_active_gases_list) {
+      // Helper fields that will temporarily store the input values
+      std::string name_transient = name + "_transient";
+      Field field_transient;
+      if (name == "co2" || name == "n2o" || name == "ch4" || name == "f11" || name == "f12") {
+        // These are all scalar (0d) time-varying
+        FieldLayout scalar0d_layout{{}, {}};
+        field_transient = create_helper_field(
+            name_transient, scalar0d_layout, m_grid->name(), ps);
```

Comment:
FWIW, I changed this check in another branch. Instead of checking the grid name, I made it to check that the field's grid name is a valid alias of the FM's grid. Not sure if it would fix the pb here.

Reply:
Also, I have been wanting to consolidate all FM's into a single FM. If I ever do that, then there will be no point in the check above, since the FM would store fields from all over the place, corresponding to whatever grids..